drivers/scheduler/k8s/k8s.go

```diff
@@ -944,6 +944,74 @@ func (k *k8s) Describe(ctx *scheduler.Context) (string, error) {
 	return buf.String(), nil
 }
 
+func (k *k8s) ScaleApplication(ctx *scheduler.Context, newScaleFactorMap map[string]int32) error {
+	logrus.Infof("Scale all Deployments")
+	k8sOps := k8s_ops.Instance()
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*apps_api.Deployment); ok {
+			dep, err := k8sOps.GetDeployment(obj.Name, obj.Namespace)
+			if err != nil {
+				return err
+			}
+			newScaleFactor := newScaleFactorMap[obj.Name+"-dep"]
+			*dep.Spec.Replicas = newScaleFactor
+			if _, err := k8sOps.UpdateDeployment(dep); err != nil {
+				return fmt.Errorf("%v", &scheduler.ErrFailedToUpdateApp{
```

Comment:
use the new error type that you defined instead of fmt.Errorf().. refer other functions in this file

Reply:
Using new error as the return type here.